addons/analytic/tests/test_analytic_account.py

```diff
@@ -122,15 +122,60 @@ def test_order_analytic_distribution_model(self):
             'partner_id': self.partner_a.id,
             'analytic_distribution': {self.analytic_account_1.id: 100, self.analytic_account_2.id: 100},
             'partner_category_id': partner_category.id,
+            'sequence': 1,
         })
 
         distribution_json = self.env['account.analytic.distribution.model']._get_distribution({
             "partner_id": self.partner_a.id,
             "company_id": self.company.id,
             "partner_category_id": partner_category.ids,
         })
-        self.assertEqual(distribution_json, distribution_4.analytic_distribution,
-                         "Distribution 4 should be given, as the partner_category_id is better than the company_id")
+
+        self.assertEqual(distribution_json, distribution_4.analytic_distribution | self.distribution_1.analytic_distribution,
+                         "Distribution 4 & 1 should be given based on sequence")
+
+    def test_distribution_model_sequence(self):
+        DistributionModel = self.env['account.analytic.distribution.model']
+        criteria = {
+            'company_id': self.company.id,
+        }
+        distribution = DistributionModel._get_distribution(criteria)
+        self.assertEqual(distribution, {}, "No model should apply, as all existing models define a partner")
+
+        criteria = {
+            'company_id': self.company.id,
+            'partner_id': self.partner_a.id,
+        }
+        distribution = DistributionModel._get_distribution(criteria)
+        self.assertEqual(distribution, self.distribution_1.analytic_distribution, "distribution_1 should apply")
```

Comment:
Does this test anything that wasn't tested before?
It all seems pretty basic :thinking: